source/common/router/retry_state_impl.h

```diff
@@ -139,6 +146,7 @@ class RetryStateImpl : public RetryState {
   uint32_t retries_remaining_{};
   uint32_t host_selection_max_attempts_;
   Upstream::ResourcePriority priority_;
+  RetrySchedulingState scheduling_state_;
```

Comment:
Needed a tiny bit of extra state to make sure all the new stats/resources were tracked correctly. This is needed to:
1. Disambiguate exponential vs ratelimited backoff on the callback or reset
2. Identify whether the retry already happened at the point when the reset happens